test/test.hpp

```diff
@@ -127,8 +127,10 @@ struct xpr_test_case
 // test_runner
 template<typename BidiIter>
 struct test_runner
-  : std::unary_function<xpr_test_case<BidiIter>, void>
 {
+    typedef BidiIter argument_type;
```

Comment:
You're so right. I hate this copy-and-paste thing so much: neither proof-reading nor any of my compilers spotted this in any of the testsuite runs.

Fixed and passes testsuite on msvc 9.0, 12.0, and 14.1